torchaudio/datasets/commonvoice.py

```diff
@@ -1,25 +1,29 @@
-import os
 import csv
+import os
 import warnings
 from pathlib import Path
 from typing import List, Dict, Tuple, Union, Optional
 
-import torchaudio
 from torch import Tensor
 from torch.utils.data import Dataset
 
+import torchaudio
+
 
 def load_commonvoice_item(line: List[str],
                           header: List[str],
                           path: str,
-                          folder_audio: str) -> Tuple[Tensor, int, Dict[str, str]]:
+                          folder_audio: str,
+                          ext_audio: str) -> Tuple[Tensor, int, Dict[str, str]]:
     # Each line as the following data:
     # client_id, path, sentence, up_votes, down_votes, age, gender, accent
 
     assert header[1] == "path"
     fileid = line[1]
-
-    filename = os.path.join(path, folder_audio, fileid)
+    if fileid.endswith(".wav"):
+        filename = os.path.join(path, folder_audio, fileid)
+    else:
+        filename = os.path.join(path, folder_audio, fileid + ext_audio)
```

Comment:
I believe CommonVoice is all `mp3` format, so checking against `.wav` would not work outside of the unit test where we intentionally change it to `wav` due to the limitation on `mp3` format support on Windows.

Since you are passing ext_audio from `CommonVoice._ext_audio`, in this implementation, you can do

```python
filename = os.path.join(path, folder_audio, field)
if not filename.endswith(ext_audio):
    filename += ext_audio
```